### PR TITLE
feat(chat): Overhaul typing indicator

### LINKED ISSUE #1054: fix(ui): Black bar appears when someone is typing to you
**Issue:**
When someone is typing to you in chat and the typing indicator appears, a black bar will stretch across the screen as well 

**OS:**
All

**Screenshot:**
image

### PR DESCRIPTION
**Pull request by Flemmli97**
### What this PR does 📖

- Overhauls the typing indicator:
    - Shows the typing indicator below the chatbar as:
    - <user> is typing / <users> are typing 
    - Multiple user are typing if more than 3 are typing

- Currently if the user(s) name(s) are too long (bigger than 17 chars) they are shortened with 3 dots.
- Despite this though it still looks a bit weird if the app is shrunk just before going into minimal view as the message in general is so "big". I am unsure how we should solve this.

If sufficient big no problem:
Screenshot 2023-08-02 at 14 23 51
If small the whole message will currently get cut off (not sure if this is fine or not)
Screenshot 2023-08-02 at 14 27 13

### Which issue(s) this PR fixes 🔨

- Resolve #1054

### CODE REVIEW

**Comment by phillsatellite** [comment]
Checked on MacOS/Linux/Windows and on each the typing indicator would appear even when other user is not typing 


https://github.com/Satellite-im/Uplink/assets/93608357/af7680a4-0dba-42f6-976f-72a3d0585140

**Comment by Flemmli97** [comment]
o. oops. left debug stuff in

File: kit/src/components/message_typing/style.scss [hunk 1]
```diff
@@ -1,19 +1,22 @@
...
 	display: inline-flex;
 	align-items: center;
-	padding: var(--padding-less) var(--padding);
+	//padding: var(--padding-less) var(--padding);
+	padding: 0 var(--padding-less);
+	padding-right: 40px;
 	width: fit-content;
 	align-self: flex-start;
-	background: var(--secondary);
+	//background: var(--secondary);
```
Comment (stavares843) [inline kit/src/components/message_typing/style.scss:16]: do we need this commented out code? 🔨

**Review by phillsatellite — APPROVED** [review]

File: kit/src/components/message_typing/mod.rs [hunk 2]
```diff
@@ -1,24 +1,63 @@
...
+        })
+    } else {
+        let mut users = cx.props.typing_users.join(", ");
+        let users = if users.len() > MAX_LEN {
+            let mut users: String = users.drain(..(MAX_LEN - 3)).collect();
+            users.push_str("...");
+            users
+        } else {
+            users
+        };
```
Comment (sdwoodbury) [inline kit/src/components/message_typing/mod.rs:29]: i think that lines 16-36 could be replaced by lines 29-36.  `join` on an array of length 1 would have no effect.

Comment (Flemmli97) [reply kit/src/components/message_typing/mod.rs:29]: done

**Review by sdwoodbury — APPROVED** [review]

**Pull request merged**